File: src/SemanticTypeForgers/forger.tests.pl
:- use_module(forger).

:- initialization(main, main).

main :-
    get_primitive_type(5, R1), R1 == 5,
    convert_to_primitive(' 123 ', R2), R2 == 123,
    convert_to_primitive('true', R3), R3 == true,
    convert_to_primitive('{}', R4), R4 == 0,

    Dict = _{productPrice: '10.5', deliveryPrice: 2.5},
    process_value(Dict, R5),
    R5 == 13.0,

    validate('PersonEmail', 'test@example.com'),
    \+ validate('PersonEmail', 'invalid-email'),

    catch(forge('PersonEmail', 'invalid-email', _), error(validation_failed('PersonEmail')), true),

    writeln('Prolog tests passed').

File: src/SemanticTypeForgers/forger_tests.erl
-module(forger_tests).
-export([start/0]).

start() ->
    5 = forger:get_primitive_type(5),
    123 = forger:convert_to_primitive(" 123 "),
    true = forger:convert_to_primitive("true"),
    0 = forger:convert_to_primitive("{}"),

    Map = #{"productPrice" => "10.5", "deliveryPrice" => 2.5},
    13.0 = forger:proccess_value("test", Map),

    true = forger:validate("PersonEmail", "test@example.com"),
    false = forger:validate("PersonEmail", "invalid-email"),

    try
        forger:forge("PersonEmail", "invalid-email"),
        erlang:error(should_have_thrown)
    catch
        error:"Validation failed for SemanticType: PersonEmail" -> ok
    end,

    io:format("Erlang tests passed~n"),
    halt(0).
